lib/puppet/util/run_mode.rb

```diff
@@ -38,6 +38,10 @@ def log_dir
         "$vardir/log"
       end
 
+      def conf_dir
+        "$puppetdir/config"
+      end
+
```

Comment:
We have a problem here because in https://github.com/puppetlabs/puppet/blob/master/lib/puppet/settings.rb#L583 we do not extrapolate the variable ( it is actually a chicken-and-egg problem because there is no settings to extrapolate yet.